bundles/org.eclipse.swt/Eclipse SWT/common/org/eclipse/swt/widgets/Synchronizer.java

```diff
@@ -63,60 +63,13 @@ public Synchronizer (Display display) {
  * @param toReceiveTheEvents the synchronizer that will receive the events
  */
 void moveAllEventsTo (Synchronizer toReceiveTheEvents) {
-	RunnableLock[] oldMessages;
-	int oldMessageCount;
-	synchronized (messageLock) {
-		oldMessages = messages;
-		messages = null;
-		oldMessageCount = messageCount;
-		messageCount = 0;
-	}
-	toReceiveTheEvents.addFirst(oldMessages, oldMessageCount);
+	messages.removeIf(toReceiveTheEvents.messages::add);
```

Comment:
I'm not sure this is functionally equivalent: this line seems to append messages from `this` into `toReceiveTheEvents.messages`; however I read (maybe erroneously) that previous behavior was appending the `toAdd` messages to the beginning of the array.
Can you please double-check, and either fix or revert if the order is changed (it seems important according to documentation contract, so better not messing too much around here).

Reply:
good finding. see https://github.com/eclipse-platform/eclipse.platform.swt/pull/85.